lib/fluxsink/a2rfluxsink.cc

```diff
@@ -0,0 +1,187 @@
+#include "globals.h"
+#include "flags.h"
+#include "fluxmap.h"
+#include "bytes.h"
+#include "protocol.h"
+#include "fluxsink/fluxsink.h"
+#include "decoders/fluxmapreader.h"
+#include "lib/fluxsink/fluxsink.pb.h"
+#include "proto.h"
+#include "fmt/format.h"
+#include "fluxmap.h"
+#include "a2r.h"
+#include <fstream>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+namespace
+{
+uint32_t ticks_to_a2r(uint32_t ticks) {
+    return ticks * NS_PER_TICK / A2R_NS_PER_TICK;
+}
+
+bool singlesided(void) {
+	return config.heads().start() == config.heads().end();
+}
+
+class A2RFluxSink : public FluxSink
+{
+public:
+	A2RFluxSink(const A2RFluxSinkProto& lconfig):
+		_config(lconfig),
+		_bytes{},
+		_writer{_bytes.writer()}
+	{
+
+		std::cout << fmt::format("A2R: writing A2R {} file containing {} tracks\n",
+			singlesided() ? "single sided" : "double sided",
+			config.cylinders().end() - config.cylinders().start() + 1
+		);
+
+		char image_date[100];
+		time_t now{std::time(nullptr)};
+		strftime(image_date, sizeof(image_date), "%FT%TZ", gmtime(&now));
```

Comment:
It _should_ be possible to do this with fmt::format --- look for strftime in https://github.com/fmtlib/fmt --- but I haven't tried it myself.